<?xml version="1.0"?>
<robot name="robot_dm">

  <link name="base_link">
   <visual>
    <geometry>
      <box size="1 1 1" />
    </geometry>
   </visual>
   <collision>
    <geometry>
      <box size="1 1 1" />
    </geometry>
   </collision>	
  </link>

  <link name="link1">
   <visual>
     <geometry>
       <cylinder length="1.5" radius="0.1" />
     </geometry>
     <origin xyz="0 0 0.75" />
   </visual>
   <collision>
     <geometry>
       <cylinder length="2.0" radius="0.1" />
     </geometry>
   </collision>	
  </link>

  <link name="link2">
   <visual>
     <geometry>
       <cylinder length="2.0" radius="0.1" />
     </geometry>
     <origin xyz="0 0 1" />
   </visual>
   <collision>
     <geometry>
       <cylinder length="1.5" radius="0.1" />
     </geometry>
   </collision>		
  </link>


  <link name="dish">
   <visual>
    <geometry>
      <cylinder length="0.02" radius="0.5" />
    </geometry>
    <origin rpy="0 0 0.01" xyz="0 0 0" />
   </visual>	
   <collision>
     <geometry>
       <cylinder length="0.02" radius="0.5" />
     </geometry>
   </collision>	
  </link>

  <joint name="base_to_link1" type="continuous">
    <parent link="base_link"/>
    <child link="link1" />
    <origin xyz="0 0 0.5" rpy="0 0 0" />
    <axis xyz="0 0 1" />
  </joint>

  <joint name="link1_to_link2" type="continuous">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0 0 1.5" rpy="0 0 0" />
    <axis xyz="1 0 0" />
  </joint>

  <joint name="link2_to_dish" type="continuous">
    <parent link="link2"/>
    <child link="dish"/>
    <origin xyz="0 0 2.0" rpy="0 0 0" />
    <axis xyz="1 0 0" />
  </joint>
</robot>
